### PR TITLE
fix: expiration time for share link when using OpenID

### PR DESCRIPTION
**Pull request by adriangitvitz**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
Using the **MINIO_BROWSER_SESSION_DURATION** env variable, we can change the token expiration created by an openid provider in order to increase the availability of the object share link, this env variable can be expressed like this:

MINIO_STS_DURATION: 2h

## Motivation and Context
https://github.com/minio/console/issues/922
https://github.com/minio/minio/issues/17902

new env variable: MINIO_STS_DURATION

## How to test this PR?

MINIO_STS_DURATION = 12h

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: internal/config/identity/openid/jwt.go [hunk 1]
```diff
@@ -181,8 +193,16 @@ func (r *Config) Validate(ctx context.Context, arn arn.ARN, token, accessToken,
 		return ErrTokenExpired
 	}
 
-	if err = updateClaimsExpiry(dsecs, claims); err != nil {
-		return err
+	timeout, exists := os.LookupEnv(config.EnvBrowserSessionDuration)
```
Comment (harshavardhana) [inline internal/config/identity/openid/jwt.go]: You need to use env.Get() API.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

File: internal/config/identity/openid/jwt.go [hunk 2]
```diff
@@ -140,6 +142,16 @@ const (
 	azpClaim = "azp"
 )
 
+func replaceClaimsExpiry(timeout string, claims map[string]interface{}) error {
```
Comment (cesnietor) [inline internal/config/identity/openid/jwt.go]: if we are never returning an error shouldn't we just not return anything?

Comment (harshavardhana) [reply internal/config/identity/openid/jwt.go]: yeah its a good idea.

**Comment by harshavardhana** [comment]
```
Running lint check
internal/config/identity/openid/jwt.go:25: File is not `gofumpt`-ed (gofumpt)
	"github.com/minio/minio/internal/config"
	"github.com/minio/pkg/v2/env"
internal/config/identity/openid/jwt.go:30: File is not `gofumpt`-ed (gofumpt)
	"time"
internal/config/identity/openid/jwt.go:31: File is not `goimports`-ed (goimports)

make: *** [Makefile:41: lint] Error 1
Error: Process completed with exit code 2.
```

File: internal/config/identity/openid/jwt.go (hunk 2 above)
Comment (adriangitvitz) [reply internal/config/identity/openid/jwt.go]: done

**Review by cesnietor — CHANGES_REQUESTED** [review]
We'll change the behavior based on an internal discussions.

File: internal/config/identity/openid/openid.go [hunk 3]
```diff
@@ -599,7 +600,12 @@ func (r Config) GetRoleInfo() map[arn.ARN]string {
 
 // GetDefaultExpiration - returns the expiration seconds expected.
 func GetDefaultExpiration(dsecs string) (time.Duration, error) {
-	defaultExpiryDuration := time.Duration(60) * time.Minute // Defaults to 1hr.
+	timeout := env.Get(config.EnvMinioStsDuration, "")
+	defaultExpiryDuration, err := time.ParseDuration(timeout)
+	//defaultExpiryDuration := time.Duration(60) * time.Minute // Defaults to 1hr.
```
Comment (cesnietor) [inline internal/config/identity/openid/openid.go]: please remove the comment

File: internal/config/identity/openid/jwt.go [hunk 4]
```diff
@@ -181,8 +192,14 @@ func (r *Config) Validate(ctx context.Context, arn arn.ARN, token, accessToken,
 		return ErrTokenExpired
 	}
 
-	if err = updateClaimsExpiry(dsecs, claims); err != nil {
-		return err
+	timeout := env.Get(config.EnvMinioStsDuration, "")
```
Comment (cesnietor) [inline internal/config/identity/openid/jwt.go]: if we already are using the env variable in updateClaimsExpiry, there's no need to do the replace here.

File: internal/config/constants.go [hunk 5]
```diff
@@ -68,6 +68,7 @@ const (
 	EnvRootDiskThresholdSize      = "MINIO_ROOTDISK_THRESHOLD_SIZE" // Deprecated Sep 2023
 	EnvBrowserLoginAnimation      = "MINIO_BROWSER_LOGIN_ANIMATION"
 	EnvBrowserSessionDuration     = "MINIO_BROWSER_SESSION_DURATION"
+	EnvMinioStsDuration           = "MINIO_STS_DURATION"
```
Comment (cesnietor) [inline internal/config/constants.go:71]: we need to pass this env variable to console, the same way we are passing MINIO_BROWSER_SESSION_DURATION, and keep suporting both with the intention to deprecate MINIO_BROWSER_SESSION_DURATION. MINIO_STS_DURATION would need to have priority.

Comment (harshavardhana) [reply internal/config/constants.go:71]: @adriangitvitz from this discussion you need to move the `if` condition around such that if `MINIO_STS_DURATION` is set it needs to override `MINIO_BROWSER_SESSION_DURATION`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/common-main.go [hunk 6]
```diff
@@ -167,9 +167,12 @@ func minioConfigToConsoleFeatures() {
 	if value := env.Get(config.EnvBrowserLoginAnimation, "on"); value != "" {
 		os.Setenv("CONSOLE_ANIMATED_LOGIN", value)
 	}
+
 	// Pass on the session duration environment variable, else we will default to 12 hours
-	if value := env.Get(config.EnvBrowserSessionDuration, ""); value != "" {
-		os.Setenv("CONSOLE_STS_DURATION", value)
+	if value_sts := env.Get(config.EnvMinioStsDuration, ""); value_sts != "" {
```
Comment (cesnietor) [inline cmd/common-main.go]: please address warnings

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
@adriangitvitz an advice before pushing the PR for CI build make sure you run `make lint` first locally as a practice to avoid triggering CI builds.

**Review by cesnietor — APPROVED** [review]
LGTM

**Pull request merged**